:bug: Source Zendesk Support: fix parse response for ticket metrics

darynaishchenko opened the pull request:
## What
https://github.com/airbytehq/oncall/issues/5078

## How
Added check for data existing  to parse response in Ticket Metrics stream.

## Review guide
<!--
1. `x.py`
2. `y.py`
-->

## User Impact
<!--
* What is the end result perceived by the user?
* If there are negative side effects, please list them. 
-->

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [ ] YES 💚
- [ ] NO ❌

erohmensing commented:
@darynaishchenko I tested this version (2.4.1-dev.420c66ebb7) compared to version 2.3.0 via our live testing tool and it seems it misses 6 records. Probably notable is these 3 log messages coming out: 
```
{"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
{"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
{"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
```

Can you take a look into this? I can send you the live testing command i used

darynaishchenko commented:
> @darynaishchenko I tested this version (2.4.1-dev.420c66ebb7) compared to version 2.3.0 via our live testing tool and it seems it misses 6 records. Probably notable is these 3 log messages coming out:
> 
> ```
> {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> ```
> 
> Can you take a look into this? I can send you the live testing command i used

@erohmensing,  Added my test results in slack thread.

darynaishchenko commented:
> > @darynaishchenko I tested this version (2.4.1-dev.420c66ebb7) compared to version 2.3.0 via our live testing tool and it seems it misses 6 records. Probably notable is these 3 log messages coming out:
> > ```
> > {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> > {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> > {"level": "ERROR", "message": "Skipping stream ticket_metrics, error message: RecordNotFound. Please ensure the authenticated user has access to this stream. If the issue persists, contact Zendesk support.", "stack_trace": null}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Can you take a look into this? I can send you the live testing command i used
> 
> @erohmensing, Added my test results in slack thread.


@erohmensing, UPD: updated TicketSubstream to avoid these log messages in case of `not found error`. I checked this version against 2.3.0v. 2.3.0 also doesn't have records for ticket ids that caused these logs. It happens when ticket metrics stream slices return ticket id for ticket that is not available now so we should skip it as metrics not available too. I ran read a couple times and for the second read these logs didn't appear.
